lib/tokenable/authable.rb

```diff
@@ -35,30 +36,35 @@ def user_class
     end
 
     def token_from_header
-      headers['Authorization'].to_s.split(' ').last
+      request.authorization.to_s.split(' ').last
     end
 
-    def token_from_user(user_id)
+    def token_from_user(user)
       jwt_data = {
-        user_id: user_id,
-      }
-      jwt_token = JWT.encode(jwt_data, jwt_secret, 'HS256')
-      {
-        user_id: user_id,
-        token: jwt_token,
+        user_id: user.id,
       }
+
+      if user.respond_to?(:issue_revoke_token)
+        jwt_data[:revoke_token] = user.issue_revoke_token
+      end
+
+      JWT.encode(jwt_data, jwt_secret, 'HS256')
     end
 
     def jwt_user_id
-      jwt['data']['user_id']
+      jwt['user_id']
```

Comment:
Much better having it in root 👍

Reply:
@marcqualie we will have to switch back to `data` when we add expiry i think.